clang/include/clang/AST/Type.h

```diff
@@ -4712,12 +4715,13 @@ class FunctionEffect {
 public:
   /// Identifies the particular effect.
   enum class Kind : uint8_t {
-    None = 0,
-    NonBlocking = 1,
-    NonAllocating = 2,
-    Blocking = 3,
-    Allocating = 4
+    NonBlocking = 0,
+    NonAllocating = 1,
+    Blocking = 2,
+    Allocating = 3,
+
```

Comment:
Would there be a value to an `EndList` sort of element, so you could use that to auto-generate `KindCount`? 

Also, is there some meaning to the values of the enums besides just wanting the default behavior of enums?

Reply:
Sure. Being explicit with enum values is an ancient habit from the days when debugging sometimes involved looking at memory in hex. I'd agree they're superfluous here.

Reply:
Although ... clangd is telling me that adding a value to the enum creates a bunch of "missing switch case" warnings. Hm.

Reply:
`Last` seems to be the most common name for that last value...

```
enum class ComparisonCategoryType : unsigned char {
  PartialOrdering,
  WeakOrdering,
  StrongOrdering,
  First = PartialOrdering,
  Last = StrongOrdering
};

enum class Lang { C = 0, CXX, LastValue = CXX };

enum class SyncScope {
  SystemScope,
  DeviceScope,
  WorkgroupScope,
  WavefrontScope,
  SingleScope,
  HIPSingleThread,
  HIPWavefront,
  HIPWorkgroup,
  HIPAgent,
  HIPSystem,
  OpenCLWorkGroup,
  OpenCLDevice,
  OpenCLAllSVMDevices,
  OpenCLSubGroup,
  Last = OpenCLSubGroup
```

Reply:
Ah, yes, we do use it quite a bit, you can also see the = <the last thing>, so just do that and use it kinda thing IMO.